packages/flutter/lib/src/material/carousel.dart

```diff
@@ -159,13 +244,16 @@ class CarouselView extends StatefulWidget {
 }
 
 class _CarouselViewState extends State<CarouselView> {
-  late double _itemExtent;
+  double? _itemExtent;
+  List<int>? _weights;
```

Comment:
Same question as below, can we just access `widget.layoutWeights` directly instead?

Reply:
Oh! I wonder then if all we need are getters instead for the controller to be able to access these? Like:

```dart
List<int> get _weights => widget.flexWeights;
```

Reply:
> The whole purpose for this is when the initial item is given some value, we know how the initial view should look like.

Is this only used for initialization? Do we need to tell the controller (i.e. in didUpdateWidget) when these values have changed on the widget so the controller can redo its calculations with the new values?

(Instead copying I would also lean toward getters, the other question as to whether we need to tell the controller when these values change still stands, though)